onnxruntime/contrib_ops/rocm/bert/tunable_op.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <iostream>
+#include <map>
+#include <vector>
+#include <string>
+#include "timer.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace rocm {
+
+struct OpParams {
+  virtual std::string signature() const = 0;
+};
+
+class Op {
+ public:
+  Op() : repeats_(100) {}
+
+  virtual void Run(const OpParams*) = 0;
+
+  void SetRepeats(int n) {
+    repeats_ = n;
+  }
+
+  float Profile(const OpParams* op_params) {
```

Comment:
do we want make the api a little flexible so we can extend to more metrics in the future? like instead of average latency, you may want latency in percental

Reply:
This would be a nice extension to add for the python profiling frontend, but I feel it might be an overkill for the purpose of selecting an op here.